tests/integration/test_load_and_run_multi_agent_a2a_tool.py

```diff
@@ -156,3 +180,121 @@ def get_datetime() -> str:
             await served_server.shutdown()
         if served_task:
             served_task.cancel()
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+@pytest.mark.asyncio
+def test_load_and_run_multi_agent_a2a_sync(
+    agent_framework: AgentFramework,
+) -> None:
+    """Tests that an agent contacts another using A2A using the adapter tool.
+
+    Note that there is an issue when using Google ADK: https://github.com/google/adk-python/pull/566
+    """
+    if agent_framework in [
+        AgentFramework.TINYAGENT,
+        AgentFramework.SMOLAGENTS,
+        AgentFramework.AGNO,
+        AgentFramework.GOOGLE,
+        AgentFramework.LLAMA_INDEX,
+    ]:
+        pytest.skip(
+            "https://github.com/mozilla-ai/any-agent/issues/357 tracks fixing so these tests can be re-enabled"
+        )
+    kwargs = {}
+
+    kwargs["model_id"] = "gpt-4.1-nano"
+    agent_model = kwargs["model_id"]
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else None
+    )
+
+    main_agent = None
+
+    tool_agent_port = 5800
+    tool_agent_endpoint = "tool_agent"
+    tool_agent_url = f"http://localhost:{tool_agent_port}/{tool_agent_endpoint}"
+
+    # DATE AGENT
+
+    date_agent_description = "Agent that can return the current date."
+    date_agent_cfg = AgentConfig(
+        instructions="Use the available tools to obtain additional information to answer the query.",
+        name="date_agent",
+        model_id=agent_model,
+        description=date_agent_description,
+        tools=[get_datetime],
+        model_args=model_args,
+    )
+    proc = multiprocessing.Process(
+        target=run_agent,
+        args=(agent_framework, date_agent_cfg, tool_agent_port, tool_agent_endpoint),
+        daemon=True,
+    )
+    proc.start()
+    while not probe(tool_agent_url):
+        time.sleep(1)
+        probe_log = f"probing <{tool_agent_url}> ..."
+        logger.info(probe_log)
```

Comment:
Is there a reason why not to serve the agent with async? I don't think you need to serve the agent sync in order to test the a2a_tool sync call, right?

Reply:
Hum. AFAICT, If you start an event loop, then you cannot use it to run to completion another async function. So, if you start by serving async'ly, then everything you do needs to be done async'ly. That's why I'm doing multiproc here; I thought about using to_thread, but probably using multiproc is easier (and already in the tests). Since we want to try a sync function, that needs to do IO, then every other IO needs to be done sync'ly in some other thread/process.

Reply:
No problem. If it needs to be done this way, no worries. Just double checking.